--- a/example.py
+++ b/example.py
@@ -1,47 +1,59 @@
 def func1(param1, param2='default val'):
-    '''Description of func with docstring javadoc style.
+    """
+    Description of func with docstring javadoc style.
 
-    @param param1: descr of param
-    @type param1: type
-    @return: some value
-    @raise KeyError: raises a key exception
+    :param param1: descr of param
+    :type param1: type
+    :param param2:  (Default value = 'default val')
+    :returns: some value
+    :raises KeyError: raises a key exception
 
-    '''
+    """
     pass
 
 def func2(param1, param2='default val2'):
-    '''Description of func with docstring reST style.
+    """
+    Description of func with docstring reST style.
 
     :param param1: descr of param
     :type param1: type
+    :param param2:  (Default value = 'default val2')
     :returns: some value
     :raises keyError: raises exception
 
-    '''
+    """
     pass
 
 def func3(param1, param2='default val'):
-    '''Description of func with docstring groups style.
+    """
+    Description of func with docstring groups style.
 
-    Params: 
-        param1 - descr of param
+    :param param1: descr of param
+    :param param2:  (Default value = 'default val')
+    :returns: some value
+    :raises keyError: raises key exception
+    :raises TypeError: raises type exception
 
-    Returns:
-        some value
-
-    Raises:
-        keyError: raises key exception
-        TypeError: raises type exception
-
-    '''
+    """
     pass
 
 class SomeClass(object):
-    '''My class.
-    '''
+    """My class."""
     def method(self, prm):
-        '''description'''
+        """
+        description
+
+        :param prm: 
+
+        """
         pass
 
     def method2(self, prm1, prm2='defaultprm'):
+        """
+        
+
+        :param prm1: 
+        :param prm2:  (Default value = 'defaultprm')
+
+        """
         pass
