--- a/issue85.py
+++ b/issue85.py
@@ -1,58 +1,76 @@
 def func(param_file, mask):
     """
-    :param lxml.etree.ElementTree param_file: xml-element tree
+
+    :param param_file: xml-element tree
         obtained from the file.
-    :param str mask: Name of mask to be considered.
+    :type param_file: lxml.etree.ElementTree
+    :param mask: Name of mask to be considered.
+    :type mask: str
+
     """
     pass
 
 
 def func2(param1: str, param2):
     """
-    :param str param1: description 1
-    :param str param2: description 2
+
+    :param param1: description 1
+    :type param1: str
+    :param param2: description 2
+    :type param2: str
     :returns: description return
+
     """
     pass
 
 
 def func3(param1: str, param2):
     """
+
     :param param1: description 1 is
         a multiline description
     :type param1: str
-    :param str param2: description 2
+    :param param2: description 2
+    :type param2: str
     :returns: description return
+
     """
     pass
 
 
 def func4(param1, param2):
     """
+
     :param param1: description 1
     :type param1: str
-    :param str param2: description 2
+    :param param2: description 2
+    :type param2: str
     :returns: description return
+
     """
     pass
 
 
 def func5(param1, param2):
     """
-    :param int param1: description 1
-    :type param1: str
+
+    :param param1: description 1
+    :type param1: int
     :param param2: description 2
     :returns: description return
+
     """
     pass
 
 
 def func6(param1: list, param2):
     """
-    :param int param1: description 1
-    :type param1: str
+
+    :param param1: description 1
+    :type param1: list
     :param param2: description 2
     :returns: description return
+
     """
     pass
 
