--- a/issue95.py
+++ b/issue95.py
@@ -3,6 +3,7 @@
 
     :param param1: 
     :returns: the return value
+
     """
     pass
 
@@ -11,7 +12,8 @@
     """Description
 
     :param param1: param's message
-    :return: the return value
+    :returns: the return value
+
     """
     pass
 
@@ -19,8 +21,9 @@
 def func3(param1):
     """Description
 
-    :param param1:
+    :param param1: 
     :returns: the return value
+
     """
     pass
 
@@ -31,6 +34,7 @@
     :param param1: 
     :param param2: 
     :returns: the return value
+
     """
     pass
 
@@ -40,7 +44,9 @@
 
     :param param1: 
     :type param1: str
+    :param param2: 
     :returns: the return value
+
     """
     pass
 
@@ -51,6 +57,7 @@
     :param param1: parameter description
     :param param2: 
     :returns: the return value
+
     """
     pass
 
@@ -58,10 +65,9 @@
 def func7(param1):
     """Description
 
-    Args:
-      param1: 
-    Returns:
-      the return value
+    :param param1: 
+    :returns: the return value
+
     """
     pass
 
@@ -70,6 +76,7 @@
     """Description
 
     :param param1: 
+
     """
     pass
 
@@ -77,7 +84,8 @@
 def func9(param1: str, param2):
     """Description
 
-    :type param1: 
+    :param param1: 
+    :type param1: str
     :param param2: 
     :returns: foobar
 
